src/main/kotlin/com/hostiflix/service/AuthenticationService.kt

```diff
@@ -87,20 +88,18 @@ class AuthenticationService (
         customerService.createCustomer(newCustomer)
     }
 
-    // SELECT * FROM auth_credentials;
     // UPDATE auth_credentials SET latest=it.latest WHERE id=it.id
-    fun setAllExistingAccessTokensToLatestFalse() {
-        val listOfAuthCredentials = authCredentialsRepository.findAll()
+    fun setAllExistingAccessTokensToLatestFalse(customerId: String) {
+        val listOfAuthCredentials = authCredentialsRepository.findAllByCustomerId(customerId)
         listOfAuthCredentials.forEach {
             it.latest = false
             authCredentialsRepository.save(it)
         }
     }
 
     // INSERT INTO auth_credentials VALUES (newAuthCredentials.id, newAuthCredentials.githubAccessToken, newAuthCredentials.customerId, newAuthCredentials.latest);
-    fun createAndStoreNewAuthCredentials(githubId: String, accessToken: String?) {
-        val customer = customerService.findCustomerByGithubId(githubId)
-        val newAuthCredentials = AuthCredentials(null, accessToken!!, customer.id!!,  true)
+    fun createAndStoreNewAuthCredentials(customerId: String, accessToken: String?) {
```

Comment:
`accessToken: String?` will always lead to a NullPointerException in case null is passed. This should not be nullable.